core/designsystem/src/main/kotlin/com/merxury/blocker/core/designsystem/component/TopAppBar.kt

```diff
@@ -163,7 +163,31 @@ fun BlockerTopAppBarNaviActionsPreview() {
     }
 }
 
-@OptIn(ExperimentalMaterial3Api::class)
+@Preview("Top App Bar with loading icon")
+@Composable
+fun BlockerTopAppBarLoadingPreview() {
```

Comment:
You may encapsulate this component, as you can see the preview did not match what it actually looks like.